src/hotspot/share/gc/g1/c1/g1BarrierSetC1.cpp

```diff
@@ -158,13 +158,13 @@ void G1BarrierSetC1::post_barrier(LIRAccess& access, LIR_Opr addr, LIR_Opr new_v
     __ logical_xor(xor_res, new_val, xor_res);
     __ move(xor_res, xor_shift_res);
     __ unsigned_shift_right(xor_shift_res,
-                            LIR_OprFact::intConst(HeapRegion::LogOfHRGrainBytes),
+                            LIR_OprFact::intConst(static_cast<jint>(HeapRegion::LogOfHRGrainBytes)),
```

Comment:
One could maybe argue that this should be a `checked_cast`, but given that the value bounds are known there is no point...